VideoLocker/src/main/java/org/edx/mobile/view/Router.java

```diff
@@ -329,4 +330,12 @@ public void showHandouts(Activity activity, EnrolledCoursesResponse courseData)
         handoutIntent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT);
         activity.startActivity(handoutIntent);
     }
+
+    public void showUserProfile(@NonNull Context context, @NonNull String username) {
+        context.startActivity(UserProfileActivity.newIntent(context, username));
```

Comment:
> Should we be delegating Intent creation to the Activity class?

Either that or at least delegate the Bundle creation. The reason being that it's the Activity class that knows what it's dependencies are, and so the logic should be kept there. The Router's purpose is to decide which Activity to launch, not to determine the implementation details of launching the activities.

> It's not the pattern for the rest of the methods here…

Yeah, we should clean them up at some point.